snakes_cafe.py

```diff
@@ -146,43 +164,91 @@ def add_to_order(order_line):
         if order_line in value.keys():
-            value[order_line][0] += 1
-            subtotal += value[order_line][1]
-            print(order_line + ' has been added. Your total is ${:.2f}'.format(subtotal * 1.101))
+            while True:
+                try:
+                    order_quantity = int(input('How many orders of ' + order_line + ' would you like?\n> '))
+                    if order_quantity > 0:
+                        update_order_quantity(order_line, order_quantity)
+                    else:
+                        print('Please enter a number between 1-' + str(value[order_line][2]))
+                    break
+                except ValueError:
+                    print('Please enter a number between 1-' + str(value[order_line][2]))
             break
     else:
         print('Please enter a valid menu item')
         return 'Please enter a valid menu item'
 
 
+def update_order_quantity(order_line, order_quantity):
+    global subtotal
+    for key, value in menu.items():
+        if order_line in value.keys():
+            if order_quantity != 0:
+                if value[order_line][2] < order_quantity:
+                    print('Oh no!! We only have ' + str(value[order_line][2]) + ' left. Please order again')
+                    add_to_order(order_line)
+                    return
+                else:
+                    value[order_line][0] += order_quantity
+                    subtotal += value[order_line][1] * order_quantity
+                    value[order_line][2] -= order_quantity
+            else:
+                value[order_line][0] += order_quantity
+                subtotal += value[order_line][1] * order_quantity
+                value[order_line][2] -= order_quantity
+    print(order_line + ' has been added. Your total is ${:.2f}\n'.format(subtotal * 1.101))
+
+
 def input_item():
     """
     changes order according to user input
     """
     global subtotal
-    order_line = input('> ').title()
+    order_line = input('What would you like?\n> ').title()
     while order_line != 'Quit':
         if order_line == 'Order':
             print_order()
         elif 'Remove' in order_line:
             remove_item(order_line)
         elif order_line == 'Menu':
             print_menu()
+        elif order_line == 'Togo':
+            optional_menu()
         elif order_line in menu:
             print_category(order_line)
         else:
             add_to_order(order_line)
-        order_line = input('> ').title()
+        order_line = input('What would you like?\n> ').title()
     print('Thank you for your order!')
     quit()
 
 
+def optional_menu():
+    """
+    allow user to use their own menu
+    """
+    filename = input('Enter your menu file: ').strip()
+
+    try:
+        #edit code block below to verify menu file is csv and can be used
+        with open(filename, 'r') as f:
+            data = f.read()
+            menu = data.split(',')
+    except:
```

Comment:
Make sure you specify an exception instead of using a generic `except`. Otherwise you will catch everything, even if it is an error you actually want to see.